src/Core/Curve25519.php

```diff
@@ -710,23 +709,23 @@ public static function fe_sq(ParagonIE_Sodium_Core_Curve25519_Fe $f)
         $f0f8_2  = self::mul($f0_2,  $f8,    25);
         $f0f9_2  = self::mul($f0_2,  $f9,    25);
         $f1f1_2  = self::mul($f1_2,  $f1,    24);
-        $f1f2_2  = self::mul($f1_2,  $f2,    25);
+        $f1f2_2  = self::mul($f1_2,  $f2,    26);
         $f1f3_4  = self::mul($f1_2,  $f3_2,  25);
         $f1f4_2  = self::mul($f1_2,  $f4,    25);
         $f1f5_4  = self::mul($f1_2,  $f5_2,  26);
         $f1f6_2  = self::mul($f1_2,  $f6,    25);
         $f1f7_4  = self::mul($f1_2,  $f7_2,  25);
         $f1f8_2  = self::mul($f1_2,  $f8,    25);
         $f1f9_76 = self::mul($f9_38, $f1_2,  25);
-        $f2f2    = self::mul($f2,    $f2,    25);
+        $f2f2    = self::mul($f2,    $f2,    26);
         $f2f3_2  = self::mul($f2_2,  $f3,    24);
         $f2f4_2  = self::mul($f2_2,  $f4,    25);
         $f2f5_2  = self::mul($f2_2,  $f5,    25);
         $f2f6_2  = self::mul($f2_2,  $f6,    25);
-        $f2f7_2  = self::mul($f2_2,  $f7,    24);
-        $f2f8_38 = self::mul($f8_19, $f2_2,  26);
-        $f2f9_38 = self::mul($f9_38, $f2,    25);
-        $f3f3_2  = self::mul($f3_2,  $f3,    24);
+        $f2f7_2  = self::mul($f2_2,  $f7,    25);
+        $f2f8_38 = self::mul($f8_19, $f2_2,  27);
+        $f2f9_38 = self::mul($f9_38, $f2,    26);
+        $f3f3_2  = self::mul($f3_2,  $f3,    25);
```

Comment:
These constants had to be increased to accommodate Ristretto255. The extra cycles shouldn't be too painful.